### PR TITLE
Add 2.7.3 docs for repo

### LINKED ISSUE #10665: [Website + Doc] Generate 2.7.3 docs
The community is preparing to announce the 2.7.2 release (almost done) and Pulsar is on the way to 2.7.3 (and 2.8.0). 

Since 2.7.3 docs might be different from 2.7.2, but now we [do not have the 2.7.3 docs](https://github.com/apache/pulsar/tree/master/site2/website/versioned_docs) and can not update the 2.7.3 docs if needed (in fact, now we have a [PR](https://github.com/apache/pulsar/pull/10638) labeled w/ 2.7.3 which needs to add docs.) 

Consequently, 2.7.3 docs need to be generated.

### PR DESCRIPTION
**Pull request by wolfstudy**
Signed-off-by: xiaolongran <xiaolongran@tencent.com>

Fixes #10665

### Motivation

Add 2.7.3 docs for repo

### Modifications

- Add 2.7.3 docs in `versioned_docs`
- Add sidebars in `versioned_sidebars`
- Update `release.json`
- Update `versions.json `

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@wolfstudy thanks for your contribution!

Seems that `release.json` and `versions.json` should be updated after 2.7.3 is released (at that time, 2.7.3 should be displayed on the website)

Take 2.7.2 for references: https://github.com/apache/pulsar/pull/10256#issuecomment-822506632 https://github.com/apache/pulsar/pull/10585

@Jennifer88huang @eolivelli please correct me if I'm wrong.

**Comment by Jennifer88huang** [comment]
@wolfstudy great job. Thank you for taking the lead to do it.
@Anonymitaet Yes, you're correct.

**Comment by wolfstudy** [comment]
@Anonymitaet @Jennifer88huang PTAL thanks.

**Comment by wolfstudy** [comment]
> @wolfstudy thanks for your contribution!
> 
> Seems that `release.json` and `versions.json` should be updated after 2.7.3 is released (at that time, 2.7.3 should be displayed on the website)
> 
> Take 2.7.2 for references:
> 
> * [#10256 (comment)](https://github.com/apache/pulsar/pull/10256#issuecomment-822506632)
> * #10585
> 
> @Jennifer88huang @eolivelli please correct me if I'm wrong.

Yes, yu. The 2.7.3 has not yet been released, so we can temporarily use it as a historical document. After 2.7.3 is released, we will show it to users as the main version of the document.

**Review by Anonymitaet — APPROVED** [review]

File: site2/website/releases.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 [
   "2.7.2",
   "2.7.1",
+  "2.7.3",
```
Comment (eolivelli) [inline site2/website/releases.json:4]: please do not add 2.7.3 here.
it has not been released

File: site2/website/versions.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 [
   "2.7.2",
   "2.7.1",
+  "2.7.3",
```
Comment (eolivelli) [inline site2/website/versions.json:4]: please do not had 2.7.3, is has not been released

**Review by eolivelli — CHANGES_REQUESTED** [review]

**Comment by eolivelli** [comment]
> The 2.7.3 has not yet been released, so we can temporarily use it as a historical document. After 2.7.3 is released, we will show it to users as the main version of the document.

I do not understand this comment, can you please explain ?
we must not list 2.7.3 in the menu

**Comment by wolfstudy** [comment]
> > The 2.7.3 has not yet been released, so we can temporarily use it as a historical document. After 2.7.3 is released, we will show it to users as the main version of the document.
> 
> I do not understand this comment, can you please explain ?
> we must not list 2.7.3 in the menu

Cool, will fix it, PTAL again.

**Review by Jennifer88huang — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]

**Pull request merged**